### PR TITLE
Converted Friends and Family into Pure Class Component.

### PR DESCRIPTION
**Pull request by hemantsain**
- Coverted Friends and Family AddressBookContent page into Pure Class Component from functional component.

### CODE REVIEW

File: src/pages/AddressBookContents.tsx [hunk 1]
```diff
@@ -42,41 +44,74 @@ import KnowMoreButton from '../components/KnowMoreButton';
...
+  isLoadContacts: boolean, 
+  selectedContact: any[],
+  loading: boolean, 
+  MyKeeper: any[],
+  IMKeeper: any[],
+  trustedContact: any[],
+  OtherTrustedContact: any[], 
+  onRefresh: boolean
+}
+class AddressBookContents extends React.PureComponent<AddressBookContentsPropTypes, AddressBookContentsStateTypes> {
```
Comment (abdulmuneer22) [inline src/pages/AddressBookContents.tsx:65]: Please import it as
import React ,{PureComponent} from 'react'

class AddressBookContents extends PureComponent .....
...

File: src/pages/AddressBookContents.tsx [hunk 2]
```diff
@@ -42,41 +44,74 @@ import KnowMoreButton from '../components/KnowMoreButton';
...
+    });
+  }
 
-  useEffect(() => {
-    setTimeout(() => {
-      setLoading(false);
-    }, 1000);
-  }, [trustedContact]);
+  componentDidUpdate(prevProps) {
+    if(prevProps.regularAccount.hdWallet.derivativeAccounts !== this.props.regularAccount.hdWallet.derivativeAccounts) {
```
Comment (abdulmuneer22) [inline src/pages/AddressBookContents.tsx:94]: Please use idx here

File: src/pages/AddressBookContents.tsx [hunk 3]
```diff
@@ -92,7 +127,7 @@ export default function AddressBookContents(props) {
           if (!contactInfo) continue;
```
Comment (abdulmuneer22) [inline src/pages/AddressBookContents.tsx:127]: Please check if contactInfo is defined

File: src/pages/AddressBookContents.tsx [hunk 4]
```diff
@@ -456,145 +420,161 @@ export default function AddressBookContents(props) {
...
+                      MyKeeper.map((item, index) => {
+                        return this.getElement(item, index, 'My Keepers');
+                      })
+                    ) : (
+                        <View style={{ height: wp('22%') + 30 }} />
+                      )}
+                  </View>
+                </View>
+              ) : (
+                  this.getWaterMark()
```
Comment (abdulmuneer22) [inline src/pages/AddressBookContents.tsx:453]: Please check this too

File: src/pages/AddressBookContents.tsx [hunk 5]
```diff
@@ -456,145 +420,161 @@ export default function AddressBookContents(props) {
...
-                      source={require('../assets/images/icons/icon_add_grey.png')}
-                    />
-                    <View>
-                      <Text style={styles.contactText}>
-                        Add Contact
-                      </Text>
-                    </View>
-                  </TouchableOpacity>
+                    >
+                      <Image
```
Comment (abdulmuneer22) [inline src/pages/AddressBookContents.tsx:509]: no inline styles please

File: src/pages/AddressBookContents.tsx (hunk 3 above)
Comment (hemantsain) [reply src/pages/AddressBookContents.tsx:127]: This is require because if contactInfo comes null then it start the process again via continue.

**Pull request merged**